plugins/funind/glob_termops.mli

```diff
@@ -122,3 +122,10 @@ val zeta_normalize : Glob_term.glob_constr -> Glob_term.glob_constr
 
 
 val expand_as : glob_constr -> glob_constr
+
+
+(* [resolve_and_replace_implicits ?expected_type env sigma rt] solves implicits of [rt] w.r.t. [env] and [sigma] and then replace them by their solution 
+ *)
+val resolve_and_replace_implicits :
```

Comment:
I agree that working at the level of `constr` also comes with drawbacks (pattern matching representation, DeBruijn surgery, etc.). I don't have enough experience to know what the intended design is in this case, but it is certainly something we should make clear for plugin writers. Maybe @ppedrot, @mattam82 and @gares have their own views?